src/arch/x86_64/CodeGen.zig

```diff
@@ -7375,12 +7442,16 @@ fn genSetReg(self: *Self, ty: Type, reg: Register, mcv: MCValue) InnerError!void
             else => switch (mcv) {
                 else => unreachable,
                 .linker_load => {
-                    try self.loadMemPtrIntoRegister(reg, Type.usize, mcv);
-                    try self.asmRegisterMemory(
-                        .mov,
-                        registerAlias(reg, abi_size),
-                        Memory.sib(Memory.PtrSize.fromSize(abi_size), .{ .base = reg.to64() }),
-                    );
+                    if (mcv.isThreadlocal()) {
+                        try self.loadThreadLocalPtrIntoRegister(reg, Type.usize, mcv);
+                    } else {
+                        try self.loadMemPtrIntoRegister(reg, Type.usize, mcv);
+                        try self.asmRegisterMemory(
+                            .mov,
+                            registerAlias(reg, abi_size),
+                            Memory.sib(Memory.PtrSize.fromSize(abi_size), .{ .base = reg.to64() }),
+                        );
```

Comment:
Same thing here, but this time we want to load a global value into a register, so once we have a pointer to the global, we need to load from it.